### PR TITLE
Add guard checks to Keyword.update/4

### PR DESCRIPTION
**Pull request by eksperimental**

### CODE REVIEW

File: lib/elixir/lib/keyword.ex [hunk 1]
```diff
@@ -925,16 +925,19 @@ defmodule Keyword do
...
   end
 
-  def update([{_, _} = e | keywords], key, default, fun) do
-    [e | update(keywords, key, default, fun)]
+  defp update_guarded([{_, _} = pair | keywords], key, default, fun) do
+    [pair | update_guarded(keywords, key, default, fun)]
   end
 
-  def update([], key, default, _fun) when is_atom(key) do
+  defp update_guarded([], key, default, _fun) when is_atom(key) do
```
Comment (josevalim) [inline lib/elixir/lib/keyword.ex:940]: We can remove the guard on this one.

File: lib/elixir/lib/keyword.ex [hunk 2]
```diff
@@ -893,8 +893,8 @@ defmodule Keyword do
     [{key, fun.(value)} | delete(keywords, key)]
   end
 
-  defp update!([{_, _} = e | keywords], key, fun, original) do
-    [e | update!(keywords, key, fun, original)]
+  defp update!([{_, _} = pair | keywords], key, fun, original) do
+    [pair | update!(keywords, key, fun, original)]
   end
 
   defp update!([], key, _fun, original) when is_atom(key) do
```
Comment (josevalim) [inline lib/elixir/lib/keyword.ex:900]: We can remove this guard too!

File: lib/elixir/lib/keyword.ex (hunk 1 above)
Comment (eksperimental) [reply lib/elixir/lib/keyword.ex]: done

File: lib/elixir/lib/keyword.ex (hunk 2 above)
Comment (eksperimental) [reply lib/elixir/lib/keyword.ex]: done

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**